### PR TITLE
[ThemeBundle] Documentation for new bundle

### PR DESCRIPTION
**Pull request by pamil**

### CODE REVIEW

File: bundles/SyliusThemeBundle/theme_configuration_reference.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Theme configuration reference
+=============================
+
+.. code-block:: json
+
+    {
+        "name": "Theme human-readable name",
+        "logical_name": "vendor/sylius-theme",
+        "description": "Optional description",
+        "parents": [
```
Comment (pjedrzejewski) [inline bundles/SyliusThemeBundle/theme_configuration_reference.rst]: Maybe ``extends``?

File: bundles/SyliusThemeBundle/theme_configuration_reference.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Theme configuration reference
+=============================
+
+.. code-block:: json
+
+    {
+        "name": "Theme human-readable name",
+        "logical_name": "vendor/sylius-theme",
```
Comment (pjedrzejewski) [inline bundles/SyliusThemeBundle/theme_configuration_reference.rst]: ``code``? Not sure, wdyt?

Comment (pamil) [inline bundles/SyliusThemeBundle/theme_configuration_reference.rst]: That was some convention in templating that uses `getLogicalName()` methods and we're decorating it with `logical_name` from the configuration.

Comment (pjedrzejewski) [inline bundles/SyliusThemeBundle/theme_configuration_reference.rst]: Ok, sounds good.

File: bundles/SyliusThemeBundle/your_first_theme.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+Private themes should be added to ``app/themes`` directory by default. Every theme should have a default confguration
+located in ``composer.json`` file. The only required parameter is ``name``, but it is worth to define other.
+
+.. code-block:: json
+
+    {
+      "name": "vendor/default-theme"
+    }
+
+When adding or removing theme, it's necessary to rebuild the container (same as adding new translation files in Symfony2) by clearing the cache (`app/console cache:clear`).
```
Comment (ahilles107) [inline bundles/SyliusThemeBundle/your_first_theme.rst]: Why exactly we need to do that? There are some new services created per theme? We would like use that bundle as a base for our theming system but in saas app clearing cache on every new theme installation is not an option ;/

**Comment by pjedrzejewski** [comment]
:dart: I missed this one, lol!

File: bundles/SyliusThemeBundle/your_first_theme.rst (hunk 3 above)
Comment (pjedrzejewski) [inline bundles/SyliusThemeBundle/your_first_theme.rst]: @ahilles107 We need it to regenerate templates paths, translations and few more things. Supporting themes loaded at runtime is something we would love to have too, but not that easy to handle... We will look at that in future versions.

**Pull request merged**